Assets/MRTK/SDK/Experimental/StepSlider/StepSlider.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.UI;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// A slider with a fixed number of step values that can be moved by grabbing / pinching a slider thumb
+    /// Number of steps defaults to 0
+    /// </summary>
+    [AddComponentMenu("MRTK/SDK/Experimental/StepSlider")]
+    public class StepSlider : PinchSlider, IMixedRealityPointerHandler
+    {
+
+        #region Serialized Fields and Properties
+        [Experimental]
+        [SerializeField]
+        [Tooltip("Number of subdvisions the slider is split into.")]
+        
+        private int sliderStepDivisions = 0;
+        /// <summary>
+        /// Property accessor of sliderStepDivisions, it holds the number of subdvisions the slider is split into.
+        /// </summary>
+        public int SliderStepDivisions
+        {
+            get 
+            { 
+                return sliderStepDivisions; 
+            }
+            set
+            {
+                sliderStepDivisions = value;
+            }
+        }
+        #endregion
+
+        #region Protected Members
+        /// <summary>
+        /// Private member used to adjust slider values
+        /// </summary>
+        protected float sliderStepVal = 0.01f;
+        #endregion
+
+        #region Unity methods
+        new void Start()
+        {
+            updateStepVal();
+            checkSliderInit();
+            base.Start();
+        }
+        #endregion
+
+        #region Private Methods
+        /// <summary>
+        /// Private method used to calculate and update the slider step divisions based on the sliderStepDivisions provided 
+        /// Slider value initialized to provided sliderInitStep value
+        /// </summary>
+        private void updateStepVal()
```

Comment:
Please use [PascalCasing](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions) on this method.